src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const result = {};
+  const strings = sourceString.split(';').map((str) => str.trim());
+
+  strings.forEach((string) => {
+    const [key, ...values] = string.split(':');
+    const value = values.join('').trim();
+    const trimmedKey = key.trim();
+
+    if (trimmedKey && value) {
+      result[trimmedKey] = value;
+    }
+  });
+
+  return result;
```

Comment:
The function correctly checks for empty keys and values before adding them to the result object. However, there's no handling of edge cases like having multiple colons in the declaration or checking for valid CSS property names.